Arrowgene.Ddon.GameServer/Characters/CraftManager.cs

```diff
@@ -0,0 +1,418 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using Arrowgene.Ddon.Shared.Entity.PacketStructure;
+using Arrowgene.Ddon.Shared.Model;
+
+namespace Arrowgene.Ddon.GameServer.Characters
+{
+    public class CraftCalculationResult
+    {
+        public uint CalculatedValue { get; set; }
+        public bool IsGreatSuccess { get; set; }
+    }
+
+    public class CraftManager
+    {
+        // TODO: introduce new assets instead
+        private static readonly List<uint> craftRankExpLimit = new()
+        {
+            0, 8, 120, 240, 400, 800, 1400, 2150, 3050, 4100, 5300, 6600, 8000, 9500, 11000, 13000, 15000, 17500, 20000, 23000, 27000, 29500, 32500, 36000, 39500, 43500, 47000,
+            51000, 55000, 60000, 63000, 67461, 72101, 76927, 81946, 87165, 92593, 98239, 104110, 110216, 116566, 123170, 130039, 137182, 144611, 152337, 160372, 168728, 177419,
+            186457, 195857, 205633, 215800, 226374, 237370, 248807, 260701, 273070, 285935, 299314, 313228, 327699, 342748, 358400, 374678, 391606, 409212, 427522, 446565, 466369
+        };
+
+        private static readonly Dictionary<uint, uint> craftRankLimitPromotionRecipes = new()
```

Comment:
I only tested the first one, but the client determines whether they need to be shown based on your current crafting rank. If you reach e.g. 8/8 it will show the relevant recipe at that level. It was extracted from the client files.